grammars/gfm.cson

```diff
@@ -1002,7 +1002,28 @@
         'include': 'source.clojure'
       }
     ]
-  },
+  }
+  {
+    'begin': '^\\s*(`{3,}|~{3,})\\s*(?=\\s*[-\\w]+\\s*$)'
+    'beginCaptures':
+      '0':
+        'name': 'support.gfm'
+    'end': '^\\s*\\1((?<=`)`+|(?<=~)~+)?\\s*$'
+    'endCaptures':
+      '0':
+        'name': 'support.gfm'
+    'name': 'markup.code.other.gfm'
+    'patterns': [
+      {
+        'begin': '\\G\\s*([-\\w]+).*$'
+        'beginCaptures':
+          '0':
+            'name': 'support.gfm'
+        'end': '^(?=\\s*[`~]{3,}\\s*$)'
```

Comment:
@50Wliu Due to the hack necessitated by the `contentName` bug, we may need to split this into two rules (one for each possible border style). The reasoning is that the inner nesting rule (`source.embedded.${1:/downcase}`) needs to terminate before the closing border, but the inner-rule has no way of knowing which border-style commenced the code-bock.

(I hope this made sense)